File: CareerDayApi/Entities/Student.cs
namespace CareerDayApi.Entities
{
    public class Student
    {
        public int Id { get; set; }
        public School School { get; set; }
        public string StudentNumber { get; set; }
        public string LastFirstName { get; set; }
        public string LastName { get; set; }
        public string FirstName { get; set; }
        public string Gender { get; set; }
        public int Grade { get; set; }
        public string Email { get; set; }
        public string HomeroomTeacher { get; set; }
        public string HomeroomNumber { get; set; }
        public int EventId { get; set; }
        public bool SurveyComplete { get; set; }
        public Event Event { get; set; }
        public List<Session> Sessions { get; set; } = [];
    }
}

File: Client/src/app/models/student.ts
import { CareerEvent } from "./event";
import { School } from "./school";

export interface Student {
    id: number,
    school: School,
    studentNumber: string,
    lastFirstName: string,
    lastName: string,
    firstName: string,
    gender: string,
    grade: number,
    email: string,
    homeroomTeacher: string,
    homeroomNumber: string,
    event: CareerEvent,
    surveyComplete: boolean
}

export interface StudentParams {
    searchTerm?: string,
    gender: string,
    grades: string[],
    surveyComplete: boolean | null,
    pageNumber: number,
    pageSize: number
}